identity/user_secret.go

```diff
@@ -0,0 +1,31 @@
+package identity
+
+import (
+	"crypto/rand"
+
+	"golang.org/x/crypto/blake2b"
+)
+
+const userSecretSize = 32
+
+func hashUserID(trustchainID []byte, userIDString string) []byte {
+	userIDBuffer := append([]byte(userIDString), trustchainID...)
+	hashedUserID := blake2b.Sum256(userIDBuffer)
+	return hashedUserID[:]
+}
+
+func createUserSecret(userID []byte) []byte {
+	randdata := make([]byte, userSecretSize-1)
```

Comment:
this is a bad function name, add a comment when you use cryptic functions such as this one